gradle-plugin/src/main/kotlin/org/mozilla/android/MozillaPlugin.kt

```diff
@@ -39,6 +48,170 @@ class MozillaPlugin : Plugin<Project> {
+                }
+            }
+        }
+
+        /**
+         * Android dependencies default to the "@aar" classifier.  Substituting dependencies "loses"
+         * the default after substitution.  We can either force consumers to specify (with "@aar")
+         * themselves or we can patch things up ourselves.  There are many consumers so we patch
+         * things up ourselves.  However, classifier specifications are second-class citizens in
+         * Gradle, so patching things is equivalent to removing the existing dependency and adding a
+         * new dependency with a non-default classifier.  Janky, but it seems to work.
+         *
+         * Why not just replace the dependency here?  We could, but it looks more mysterious.  In
+         * the output of `dependencies` tasks, for instance, doing this patching up and then
+         * substituion gives helpful output showing `before -> after` mappings.
+         */
+        fun forceGeckoViewDependenciesToHaveAarClassifier(project: Project): Boolean {
+            var anyRemoved = false
+            var anyAdded = false
+
+            project.configurations.all { config ->
+                if (config.isCanBeResolved()) {
+                    val seen: MutableSet<Configuration> = mutableSetOf()
+                    val next: MutableList<Configuration> = mutableListOf(config)
+                    while (next.isNotEmpty()) {
+                        val c = next.removeAt(0)
+                        if (!seen.add(c)) {
+                            continue
+                        }
+                        next.addAll(c.extendsFrom.toList())
+
+                        val toRemove = c.dependencies.filter { dep -> dep.group == "org.mozilla.geckoview" && dep.name.startsWith("geckoview-nightly") }
+                        for (d in toRemove) {
+                            val removed = c.dependencies.remove(d)
+                            val added = c.dependencies.add(project.dependencies.create("${d.group}:${d.name}:${d.version}@aar"))
+                            pluginLogger.debug("Forced geckoview dependency to have @aar classifier in ${c} (inherited from ${config})")
+
+                            anyRemoved = anyRemoved || removed
+                            anyAdded = anyAdded || added
+                        }
+                    }
+                }
+            }
+
+            return (anyRemoved || anyAdded)
+        }
+
+        /**
+         * If the given configuration consumes GeckoView dependencies, substitute for the local
+         * GeckoView dependency instead.
+         */
+        fun substituteGeckoViewDependenciesForLocalGeckoView(config: Configuration, arch: String) {
+            if (!config.isCanBeResolved()) {
+                return
+            }
+
+            config.resolutionStrategy { strategy ->
+                strategy.dependencySubstitution.all inner@{dependency ->
```

Comment:
The IDE shows a warning this API (`dependencySubstitution`) is unstable. I assume we'd want to bump to Gradle 5+ soon: will it still work?